File: web/src/features/reconcile/api/reconcile.ts
import { api } from '@shared/api';

// Reconcilable resource kinds, matching the server's reconcile registry
// (GET /api/v1/reconcile/kinds). For kind 'agent', `name` is the instance UID.
export type ReconcileKind = 'agent' | 'agentgroup' | 'agentremoteconfig';

// reconcileResource asks the server to re-run the side effects that normally fire when the
// named resource is created/updated, re-enforcing its domain rules on demand.
export function reconcileResource(
  kind: ReconcileKind,
  namespace: string,
  name: string,
): Promise<void> {
  return api.post(`/api/v1/namespaces/${namespace}/reconcile/${kind}/${encodeURIComponent(name)}`);
}

File: pkg/apiserver/application/usecase/reconcile.go
package usecase

import "context"

// ReconcileManageUsecase re-enforces a domain object's invariants on demand. It dispatches a
// (kind, namespace, name) request to the reconciler registered for that kind, re-running the
// side effects that normally fire when the resource is created or updated. New reconcilable
// kinds become available here without changing this interface.
type ReconcileManageUsecase interface {
	// Reconcile re-enforces the named resource's invariants. For namespace-scoped kinds name
	// is the resource name; for the "agent" kind it is the instance UID. An unknown kind or a
	// missing resource surfaces as an error mapped to 4xx by the transport.
	Reconcile(ctx context.Context, kind string, namespace string, name string) error
	// ReconcileKinds returns the reconcilable kinds, for discovery and CLI completion.
	ReconcileKinds(ctx context.Context) []string
}
